spinalcordtoolbox/reports/qc2.py

```diff
@@ -307,12 +296,52 @@ def sct_process_segmentation(
-        img = mosaic(img_seg, centers, radius)
+
+        img_temp = img_seg.copy()
+        img_temp.data = np.full_like(img_seg.data, np.nan)
+        # Look across slices, and for each slice, and for each slice
+        # Note #1: as metrics['centermass_x'] and ['centermass_y'] coordinates were obtained from RPI oriented image,
+        #  we need to convert them to the SAL orientation (used in the QC report). This is done using the hacky code
+        #  below :-D
+        # Note #2: metrics['centermass_x'] and ['centermass_y'] seem to be the same as `centers`, but they are obtained differently:
+        #   - metrics['centermass_x'] and ['centermass_y'] are obtained using PCA via spinalcordtoolbox.registration.algorithms.compute_pca
+        #   - centers is obtained using scipy.ndimage.center_of_mass
+        #  I'm using PCA-based metrics['centermass_x'] and metrics['centermass_y'] here to be consistent with code in
+        #  spinalcordtoolbox.process_seg as PCA is used there for the HOG angle computation.
+        # for i, (x, y) in enumerate(zip(metrics['centermass_x'].data, metrics['centermass_y'].data),1):
+        for i in range(img_temp.dim[0]):
+            # Center of mass
+            if 'centermass_x' in metrics and 'centermass_y' in metrics:
+                x = metrics['centermass_x'].data[i] if 'centermass_x' in metrics else np.nan
+                y = metrics['centermass_y'].data[i] if 'centermass_y' in metrics else np.nan
+                if not np.isnan(x) and not np.isnan(y):
+                    x = rpi_slice_to_orig_orientation(img_temp.dim, 'SAL', x, 2)
+                    y = rpi_slice_to_orig_orientation(img_temp.dim, 'SAL', y, 0)
+                    img_temp.data[img_temp.dim[0]-1-i, int(y), int(x)] = 1
+            # Line for the HOG angle
+            if 'angle_hog' in metrics:
+                angle_rad = metrics['angle_hog'].data[i] if 'angle_hog' in metrics else np.nan
+                # Only draw the line if x, y, and angle_rad are all valid (not NaN)
+                if not np.isnan(x) and not np.isnan(y) and not np.isnan(angle_rad):
+                    # Compute the end points of the line
+                    x_start = int(x - radius[0] * np.sin(angle_rad))
+                    y_start = int(y - radius[1] * np.cos(angle_rad))
+                    x_end = int(x + radius[0] * np.sin(angle_rad))
+                    y_end = int(y + radius[1] * np.cos(angle_rad))
+                    # Create coordinates for the line
+                    rr, cc = draw_line(y_start, x_start, y_end, x_end)
+                    # Filter coordinates to ensure they're within image bounds
+                    valid_idx = (0 <= rr) & (rr < img_temp.dim[1]) & (0 <= cc) & (cc < img_temp.dim[2])
+                    # Set line pixels to 1
+                    if np.any(valid_idx):
+                        img_temp.data[img_temp.dim[0] - 1 - i, rr[valid_idx], cc[valid_idx]] = 1
+
+        img = mosaic(img_temp, centers, radius)
```

Comment:
Okay, this works, but the line is pixelated.

<img width="1027" height="816" alt="image" src="https://github.com/user-attachments/assets/e047e72f-0a53-4985-ab2f-d5e3094fdb29" />